### DISCUSSION TITLE
Dock Plugin SizePolicy Not Functioning

### DISCUSSION THREAD

**Discussion by ZhongYic00** [linuxdeepin/developer-center#3235] [Q&A | 问答板块]
I'm building a dock plugin with customized size(90x40). As mentioned in recent dock updates, I override `pluginSizePolicy` to always return `PluginsItemInterface::Custom`. However the plugin widget is still always clipped by chance, and through output debugging/ gdb breakpoints it seems that pluginSizePolicy is not even called when determining container margins
![image](https://user-images.githubusercontent.com/38401412/172758406-175cddfe-aba6-4962-9974-5ca940facc1c.png)
Here's gdb screenshot, if pluginSizePolicy is called it'll qWarning something but it didn't.
![image](https://user-images.githubusercontent.com/38401412/172758840-bcc4c36c-a0fa-4144-8753-c4d9931f0306.png)
Deb package version is
![image](https://user-images.githubusercontent.com/38401412/172759106-6a2fde67-2def-4ee5-8612-050ca9c9a454.png)
Is the newly updated fix not packaged into this version? Or it's my overriding problem?

**Comment by justforlxz** [comment]
Can you provide the code for the plugin?

**Reply by ZhongYic00** [reply]
https://github.com/ZhongYic00/digitalwellbeing-plugin
Only cpp/h files now, sym links are to be processed later.
Thanks!
